core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/StopStartUpscaleFlowEventChainFactory.java

```diff
@@ -59,15 +54,15 @@ private void addStackSyncTriggerEvent(StackAndClusterUpscaleTriggerEvent event,
 
     private void addClusterScaleTriggerEventIfNeeded(StackAndClusterUpscaleTriggerEvent event, StackView stackView, ClusterView clusterView,
             Queue<Selectable> flowEventChain) {
-        HostGroup hostGroup = hostGroupService.getByClusterIdAndName(clusterView.getId(), event.getInstanceGroup())
-                .orElseThrow(NotFoundException.notFound("hostgroup", event.getInstanceGroup()));
-        flowEventChain.add(
-                new StopStartUpscaleTriggerEvent(
-                        StopStartUpscaleEvent.STOPSTART_UPSCALE_TRIGGER_EVENT.event(),
-                        stackView.getId(),
-                        hostGroup.getName(),
-                        event.getAdjustment(),
-                        event.getClusterManagerType())
-        );
+        for (Map.Entry<String, Integer> hostGroupWithAdjustment : event.getHostGroupsWithAdjustment().entrySet()) {
```

Comment:
This isn't meant to handle multiple hostGroups right now. Better to add a precondition check, and fail?